pkg/scheduler/actions/common/solvers/scenario/by_node_scenario.go

```diff
@@ -7,24 +7,27 @@ import (
 	"golang.org/x/exp/maps"
 	"golang.org/x/exp/slices"
 
+	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api"
 	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/common_info"
 	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/pod_info"
 	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/podgroup_info"
 	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/framework"
 )
 
+var _ api.ScenarioInfo = &ByNodeScenario{}
```

Comment:
is this needed?

Reply:
I like using [interface guards](https://rednafi.com/go/interface_guards/), it's a matter of taste, but I like to make things explicit